js/NewsChannelOperations.js

```diff
@@ -0,0 +1,44 @@
+function SaveEmail(){
+  var emailAddresss= document.getElementById('txtEmailAddress').value;
+  if (ValidateEmail(emailAddresss)){
+    localStorage.setItem('email',emailAddresss);
+  }
+}
+function ValidateEmail(mail)
+{
+  if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(mail))
+  {
+    return true;
+  }
+}
+function ShowHideDiv(selectedElement) {
+  if(selectedElement.value> 0){
+    for(var index =1 ;index< newsChannels.length; index++){
+
+      if (selectedElement.value == index){
+        document.getElementById('Channel' +index ).setAttribute('style','display:block');
+      }
+      else{
+        document.getElementById('Channel' +index).setAttribute('style','display:None');
+      }
+    }}
+    else {
+      for(var index =1 ;index< newsChannels.length; index++){
```

Comment:
use getElementsByClassName() and foreach instead.

Reply:
updated code accordingly